src/overlays/actors/ovl_En_Horse/z_en_horse.c

```diff
@@ -28,370 +165,4602 @@ const ActorInit En_Horse_InitVars = {
+        }
+    } else {
+        temp_f0 = (this->actor.scale.y * 100.0f);
+
+        if ((horseGameCheck == NULL) || (horseGameCheck->actor.id != ACTOR_EN_HORSE_GAME_CHECK) ||
+            ((horseGameCheck->actor.params & 0xFF) != 5)) {
+            if ((movingFast == false) && ((19.0f * temp_f0) < obstacleHeight) &&
+                (obstacleHeight <= (40.0f * temp_f0))) {
+                EnHorse_Stub1(this);
+                this->postDrawFunc = EnHorse_LowJumpInit;
+            } else if ((movingFast == true) &&
+                       (((this->actor.speedXZ < 13.8f) && ((19.0f * temp_f0) < obstacleHeight) &&
+                         (obstacleHeight <= (72.0f * temp_f0))) ||
+                        ((this->actor.speedXZ > 13.8f) && (obstacleHeight <= (112.0f * temp_f0))))) {
+                EnHorse_Stub2(this);
+                this->postDrawFunc = EnHorse_HighJumpInit;
+            }
+        } else if (movingFast == false) {
+            EnHorse_Stub1(this);
+            this->postDrawFunc = EnHorse_LowJumpInit;
+        } else if (movingFast == true) {
+            EnHorse_Stub2(this);
+            this->postDrawFunc = EnHorse_HighJumpInit;
+        }
+    }
+}
+
+void func_80886C00(EnHorse* this, GlobalContext* globalCtx) {
+    Input* input = &globalCtx->state.input[this->unk_52C];
+
+    if (((this->action == ENHORSE_ACT_MOUNTED_GALLOP) || (this->action == ENHORSE_ACT_MOUNTED_REARING) ||
+         (this->action == ENHORSE_ACT_STOPPING)) &&
+        (CHECK_BTN_ALL(input->press.button, BTN_A) || (func_801A5100() == 5)) &&
+        (globalCtx->interfaceCtx.unk_212 == 8) && !(this->stateFlags & ENHORSE_BOOST) &&
+        !(this->stateFlags & ENHORSE_FLAG_8) && !(this->stateFlags & ENHORSE_FLAG_9)) {
+        if (this->numBoosts > 0) {
+            func_8013ECE0(0.0f, 180, 20, 100);
+            this->stateFlags |= ENHORSE_BOOST;
+            this->stateFlags |= ENHORSE_FIRST_BOOST_REGEN;
+            this->stateFlags |= ENHORSE_FLAG_8;
+            this->numBoosts--;
+            this->boostTimer = 0;
+            if (this->numBoosts == 0) {
+                this->boostRegenTime = 140;
+            } else if (this->type == HORSE_EPONA) {
+                if (this->stateFlags & ENHORSE_FIRST_BOOST_REGEN) {
+                    this->boostRegenTime = 60;
+                    this->stateFlags &= ~ENHORSE_FIRST_BOOST_REGEN;
+                } else {
+                    this->boostRegenTime = 8;
+                }
+            } else {
+                this->boostRegenTime = 70;
+            }
+        } else if ((this->stateFlags & ENHORSE_DRAW) && (Rand_ZeroOne() < 0.1f)) {
+            if (this->type == HORSE_2) {
+                func_8019F1C0(&this->unk_218, NA_SE_EV_KID_HORSE_NEIGH);
+            } else {
+                func_8019F1C0(&this->unk_218, NA_SE_EV_HORSE_NEIGH);
+            }
+        }
+    }
+}
+
+void EnHorse_RegenBoost(EnHorse* this, GlobalContext* globalCtx) {
+    s32 playSound;
```

Comment:
```suggestion
    s32 playSfx;
```
Could you rename this?